geom/geom_prodTgtTube.txt

```diff
@@ -20,38 +22,44 @@ bool g4.placePV                = true;
-string world.materialName        = "G4_Galactic";
+string  world.materialName        = "G4_Galactic";
 
 // make the world boundaries visible
 bool   world.boxVisible          = true;
 // make the volume transparent
 bool   world.boxSolid            = false;
 
-double world.halfLength          = 10.0;
-double world.outerLayerThickness = 0.001;// the world "outer layer" to force a step point
+double world.halfLength          = 500.0;
+double world.outerLayerThickness = 0.01;// the world "outer layer" to force a step point
+
 
 // the target (a tube) for mu2e.studyEnvVersion == 1
 
 bool           tube.visible = true;
 bool           tube.solid   = false;
+// setup for the number of tubes
+int 	       tube.numberOfTubes = 19;
+
 
 // fixme: may need to use pure isotopes
-string         tube.wallMaterialName = "G4_W";
 
-double         tube.rIn        = 0.0;
-double         tube.rOut       = 1.0;
-double         tube.halfLength = 3.0;
-double         tube.phi0       = 0.;
-double         tube.phiSpan    = 360.; // degrees
-double         tube.phiRotZ    =  0.;
-double         tube.sign       =  1.; // for additional 180 deg rotation tests around Y axis (when -1)
+vector<string> tube.wallMaterialName  = {"G4_W","G4_W","G4_W","G4_W","G4_W","G4_W","G4_W","G4_W","G4_W","G4_W","G4_W","G4_W","G4_W","G4_W","G4_W","G4_W","G4_W","G4_W","G4_W","G4_W"};
+vector<double> tube.rIn        = {0,0.63, 0.75, 0.75, 0.75, 0.75, 0.63, 0.75, 0.75, 0.75, 0.75, 1.25, 1.25, 1.25, 1.25, 1.25, 1.25, 1.25, 1.25};
+vector<double> tube.rOut       = {0.25, 0.75, 1.25, 1.25, 1.25, 1.25, 0.75, 1.25, 1.25, 1.25, 1.25, 1.5, 1.5, 1.5, 1.5, 1.5, 1.5, 1.5, 1.5};
+vector<double> tube.halfLength = {6, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.5, 0.5, 0.5, 0.5, 6, 6, 6, 6};// half the height 
+vector<double> tube.phi0       = {0, 0, 352.5, 82.5, 172.5, 262.5, 0, 352.5, 82.5, 172.5, 262.5, 0, 0, 0, 0, 351.0, 81.0, 171.0, 261.0};
+vector<double> tube.phiSpan    = {360, 360, 15, 15, 15, 15, 360, 15, 15, 15, 15, 360, 360, 360, 360, 18, 18, 18, 18}; // degrees
+vector<double> tube.phiRotZ    = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
+vector<double> tube.sign       = {1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1}; // for additional 180 deg rotation tests around Y axis (when -1)
 
-int            tube.copyNumber = 3; // non 0 for volume tracking purposes
+vector<int>    tube.copyNumber = {3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3}; // non 0 for volume tracking purposes
                                     // 0 & 1 are World and an auxiliary
                                     // volume almost filling up the world
 
 // tube position in the world
-vector<double> tube.centerInWorld = {0., 0., 10.};
+vector<string>   tube.centerInWorld = { "0.0, 0.0, 0.0", "0.0, 0.0, 2.0", "0.0, 0.0, 2.0",  "0.0, 0.0, 2.0", "0.0, 0.0, 2.0", "0.0, 0.0, 2.0", "0.0, 0.0, -2.0", "0.0, 0.0, -2.0", "0.0, 0.0, -2.0", "0.0, 0.0, -2.0", "0.0, 0.0, -2.0",
```

Comment:
Can you remind me why this is a vector<string> instead of vector of arrays of length 3? Was it just a matter of compatibility with existing capabilities of the Offline? Every element in the vector will consist of exactly 3 components, so a :vector of arrays would make more sense than a vector of strings

Reply:
The offline currenntly cannot take in parameters of 2 dimensionals vectors given that the methods are not in SimpleConfig. Instead, vector string can be taken in and a function in the cc file traverses the string to conver to a vector of arrays.

Reply:
I do not believe it does unless a vector han be retreived by using the Hep3Vector implented in the offline. However, this may require adding it into SimpleConfig. I did try to make the changes before which did not work but this may have more or less to do with the iteration of 2Dimensional vectors. At the time I attempted this, I did not use the gdb debugger yet. I may reconsider using it to add the support of 2D vectors if necesarry.

Reply:
Ok, I think the path forward then is to live with this less-than-ideal implementation of a vector string for the center coordinates for each cylinder for now and additionally file an Issue in this repo saying we want to fix this once SimpleConfig allows more functionality. You could always contribute to that at a later time, if you want. Let me know if you disagree.

As a note, I suspect what we want is already possible by passing a std::vector of pointers to C arrays. However, this also feels like a kludge. A dedicated implementation of std::vector< std::array<double, 3> > would be better, but, I don't see that functionality in SimpleConfig.

Reply:
Ok, I'll resolve this conversation then. I will create an issue after this PR